graphix/sim/tensornet.py

```diff
@@ -488,28 +522,32 @@ def get_basis_amplitude(self, basis, **kwagrs):
         coef = self.get_basis_coefficient(basis, **kwagrs)
         return abs(coef) ** 2
 
-    def to_statevector(self, indices=None, **kwagrs):
-        """Retrieve the statevector from the tensornetwork.
-        This method tends to be slow however we plan to parallelize this.
+    def to_statevector(self, skip=False, path_info=False, **kwagrs):
+        """Take outer product of the tensors in the network and return the statevector.
 
         Parameters
         ----------
-        indices (optional): list of int
-            target qubit indices. Default is the MBQC output nodes (self.default_output_nodes).
+        skip : bool
+            if True, skip the simplification process.
+        path_info : bool
+            if True, return the path information of the contraction.
 
         Returns
         -------
         numpy.ndarray :
             statevector
         """
-        if indices == None:
-            n_qubit = len(self.default_output_nodes)
-        else:
-            n_qubit = len(indices)
-        statevec = np.zeros(2**n_qubit, np.complex128)
-        for i in range(len(statevec)):
-            statevec[i] = self.get_basis_coefficient(i, normalize=False, indices=indices, **kwagrs)
-        return statevec / np.linalg.norm(statevec)
+        if skip:
+            tn = self.copy()
+            output_inds = [self._dangling[str(index)] for index in self.default_output_nodes]
+            psi = tn.contract(output_inds=output_inds, **kwagrs).data.reshape(-1)
+            return Statevec(psi=psi)
+
+        tn = self.copy()
+        tn_simplified = tn.full_simplify("ADCR")
+        output_inds = [self._dangling[str(index)] for index in self.default_output_nodes]
+        psi = tn_simplified.contract(output_inds=output_inds, **kwagrs)
+        return psi if path_info else psi.data.reshape(-1)
```

Comment:
`path_info` is the flag to get [opt_einsum.contract.PathInfo](https://optimized-einsum.readthedocs.io/en/stable/autosummary/opt_einsum.contract.PathInfo.html)-like object without contraction. Altough I originally implemented it, in my reference code, to evaluate the contraction cost for the case it is impossible to explicitly calculate, this argument may be confusing for users. So, you can remove it